tests/unit/stimulus/text_test.py

```diff
@@ -0,0 +1,163 @@
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+from pathlib import Path
+
+import pytest
+
+import pymovements as pm
+
+
+@pytest.mark.parametrize(
+    ('aoi_file'),
+    [
+        pytest.param(
+            'tests/files/toy_text_1_1_aoi.csv',
+            id='toy_text_1_1_aoi',
+        ),
+        pytest.param(
+            'tests/files/toy_text_2_2_aoi.csv',
+            id='toy_text_2_2_aoi',
+        ),
+        pytest.param(
+            'tests/files/toy_text_3_3_aoi.csv',
+            id='toy_text_3_3_aoi',
+        ),
+    ],
+)
+def test_str_aoi_path(aoi_file):
+    pm.stimulus.text.from_file(
+        aoi_file,
+        character_column='char',
+        pixel_x_column='top_left_x',
+        pixel_y_column='top_left_y',
+        width_column='width',
+        height_column='height',
+        page_column='page',
+    )
+
+
+@pytest.mark.parametrize(
+    ('aoi_file'),
+    [
+        pytest.param(
+            'tests/files/toy_text_1_1_aoi.csv',
+            id='toy_text_1_1_aoi',
+        ),
+        pytest.param(
+            'tests/files/toy_text_2_2_aoi.csv',
+            id='toy_text_2_2_aoi',
+        ),
+        pytest.param(
+            'tests/files/toy_text_3_3_aoi.csv',
+            id='toy_text_3_3_aoi',
+        ),
+    ],
+)
+def test_str_aoi_path_kwargs(aoi_file):
+    pm.stimulus.text.from_file(
+        aoi_file,
+        character_column='char',
+        pixel_x_column='top_left_x',
+        pixel_y_column='top_left_y',
+        width_column='width',
+        height_column='height',
+        page_column='page',
+        custom_read_kwargs={
+            'separator': ',',
+        },
+    )
+
+
+@pytest.mark.parametrize(
+    ('aoi_file'),
+    [
+        pytest.param(
+            Path('tests/files/toy_text_1_1_aoi.csv'),
+            id='toy_text_1_1_aoi',
+        ),
+        pytest.param(
+            Path('tests/files/toy_text_2_2_aoi.csv'),
+            id='toy_text_2_2_aoi',
+        ),
+        pytest.param(
+            Path('tests/files/toy_text_3_3_aoi.csv'),
+            id='toy_text_3_3_aoi',
+        ),
+    ],
+)
+def test_Path_aoi_path(aoi_file):
+    pm.stimulus.text.from_file(
+        aoi_file,
+        character_column='char',
+        pixel_x_column='top_left_x',
+        pixel_y_column='top_left_y',
+        width_column='width',
+        height_column='height',
+        page_column='page',
+    )
+
+
+@pytest.mark.parametrize(
+    ('aoi_file'),
+    [
+        pytest.param(
+            Path('tests/files/toy_text_1_1_aoi.csv'),
+            id='toy_text_1_1_aoi',
+        ),
+        pytest.param(
+            Path('tests/files/toy_text_2_2_aoi.csv'),
+            id='toy_text_2_2_aoi',
+        ),
+        pytest.param(
+            Path('tests/files/toy_text_3_3_aoi.csv'),
+            id='toy_text_3_3_aoi',
+        ),
+    ],
+)
+def test_Path_aoi_path_kwargs(aoi_file):
```

Comment:
I think you can combine all these four test functions into one by using parametrize over the args/kwargs you passt to `from_file()`